### PR TITLE
Complete restricted-output fundrawtransaction edge coverage

### LINKED ISSUE #81: Complete restricted-output fundrawtransaction edge coverage and launch verification
## Summary

The `1.0.0` target contains the production guard introduced by
Qbit-Org/qbit#37.

`fundrawtransaction` preserves the original decoded outputs and validates each
`scriptPubKey` against `Consensus::IsRestrictedOutputModeConsensusOutput()` at
next-block height before wallet funding. Existing tests cover rejection of
P2PKH, P2SH-SegWit, P2WPKH, and P2TR outputs and acceptance of P2MR, OP_RETURN,
and PayToAnchor.

This issue tracks the remaining edge coverage and launch-verification work.

## Remaining test coverage

Extend `wallet_p2mr.py` to cover:

- P2WSH output rejection.
- A mixed P2MR/non-P2MR output set, proving one invalid output rejects the
  entire request.
- A non-extractable, non-exempt script.
- A reserved-witness output, asserting the intended wallet policy result.
- An invalid output referenced by `subtractFeeFromOutputs`.
- Decoding successful P2MR, OP_RETURN, and PayToAnchor results and confirming
  that their original scripts are preserved.

All rejection cases must fail before wallet funding returns transaction hex or
locks newly selected inputs.

## Validation

Run:

    ulimit -n 10240
    build/test/functional/test_runner.py wallet_p2mr.py wallet_fundrawtransaction.py
    build/bin/test_qbit --run_test=wallet_tests
    build/bin/test_qbit --run_test=wallet_rpc_tests

Also rerun or explicitly schedule bounded launch-target verification against
the intended release commit.

## Acceptance criteria

- The launch target is recorded as containing the production fix from
  Qbit-Org/qbit#37.
- All edge cases above have deterministic functional coverage.
- Existing allowed-output tests continue to pass.
- The review tracker is updated with the target commit and validation evidence.

## Consensus impact

Consensus-adjacent, not consensus-changing.

This work must not alter the restricted-output consensus classifier or
validation rules. It verifies that the wallet RPC fails early for transactions
those existing rules would reject. Mempool and block validation already reject
restricted-output-invalid transactions, so there is no identified
consensus-split path.

### PR DESCRIPTION
**Pull request by kiwidream**
## Summary

- extend restricted-output `fundrawtransaction` rejection coverage to P2WSH, non-extractable scripts, mixed valid/invalid outputs, and reserved witness outputs
- verify an invalid fee-subtraction request fails before locking wallet inputs
- decode successful P2MR, OP_RETURN, and PayToAnchor funding results and assert the original scripts and values are preserved

Fixes #81.

## Testing

- [x] Built locally.
  - `cmake -B build -G Ninja -DBUILD_GUI=OFF -DBUILD_TESTS=ON -DBUILD_BENCH=OFF -DBUILD_FUZZ_BINARY=OFF -DENABLE_IPC=OFF -DWITH_ZMQ=OFF -DWITH_USDT=OFF`
  - `cmake --build build --target qbitd qbit-cli test_qbit -j 10`
- [x] Ran focused unit or functional tests for the changed area.
  - `python3 -m py_compile test/functional/wallet_p2mr.py`
  - `ulimit -n 10240; build/test/functional/test_runner.py wallet_p2mr.py`
  - `ulimit -n 10240; build/test/functional/test_runner.py --timeout-factor=4 wallet_fundrawtransaction.py`
  - `ulimit -n 10240; build/bin/test_qbit --run_test=wallet_tests`
  - `ulimit -n 10240; build/bin/test_qbit --run_test=wallet_rpc_tests`
- [x] Ran lint or formatting checks relevant to this change.
  - `git diff --check`
  - Docker lint image: all checks passed.
- [ ] Not run. Reason: N/A; all planned checks were run.

## Target Branch

- [x] This PR targets `main` or a maintainer-requested release branch such as `0.1.x`.

## Risk / Review Notes

- [ ] Consensus, script, crypto, wallet, P2P, release, CI, or security-sensitive behavior changed.
- [x] No consensus, script, crypto, wallet, P2P, release, CI, or security-sensitive behavior changed.

Notes: This is a test-only, consensus-adjacent change. It exercises existing wallet preflight behavior against the existing restricted-output consensus classifier without changing transaction validity, mempool or block validation, activation boundaries, or wallet production code.

## Docs / Process Impact

Choose exactly one:

- [ ] I updated public docs because this PR changes user-visible behavior, integration guidance, release/process guidance, or expected validation.
- [x] No public docs update needed. Reason: This PR only expands regression coverage for existing behavior.

## libbitcoinpqc Subtree Checklist (if `src/libbitcoinpqc` changed)

Not applicable; `src/libbitcoinpqc` is unchanged.

- [ ] Source commit is reachable from an immutable release tag in `Qbit-Org/qbit-libbitcoinpqc`.
- [ ] qbit imports the tagged upstream tree directly without pruning or a curated subtree branch.
- [ ] Subtree import/update was performed with `contrib/devtools/update-libbitcoinpqc-subtree.sh`.
- [ ] `test/lint/libbitcoinpqc-subtree-check.sh` passes locally.
- [ ] Any default tag change in `contrib/devtools/update-libbitcoinpqc-subtree.sh` is intentional and matches `doc/subtrees/libbitcoinpqc.md`.

### CODE REVIEW

File: test/functional/wallet_p2mr.py [hunk 1]
```diff
@@ -487,16 +493,60 @@ def run_test(self):
...
+            raw_tx_with_outputs([(reserved_witness_script(3), COIN)]),
+        )
+
+        self.log.info("Check invalid fee-subtraction outputs fail before locking wallet inputs")
+        invalid_raw_tx = raw_tx_with_outputs([(invalid_p2pkh_script, COIN)])
+        locked_before = p2mr_wallet.listlockunspent()
+        for options in (
+            {"subtractFeeFromOutputs": [0], "lockUnspents": True},
+            {"subtract_fee_from_outputs": [0], "lock_unspents": True},
+        ):
```
[the remark this answers is not in the document]
Comment (kiwidream) [reply test/functional/wallet_p2mr.py]: Agreed. `fundrawtransaction` only consumes the documented `subtractFeeFromOutputs` option when building recipients, so the snake-case case did not exercise fee subtraction. Removed that ineffective case in 474342eac8 and narrowed the PR and backing issue wording accordingly. The remaining assertion verifies that the documented fee-subtraction request fails before locking wallet inputs. `wallet_p2mr.py` and Docker lint pass.
